File: memory-api/src/main/java/com/memory/dto/todo/TodoRequest.java
package com.memory.dto.todo;

import com.memory.domain.member.Member;
import com.memory.domain.todo.Todo;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;

import java.time.LocalDateTime;

public class TodoRequest {

    @Getter
    public static class Create {
        @NotBlank(message = "제목은 필수 입력값입니다.")
        private String title;

        private final String content;

        @NotNull(message = "마감일시는 필수 입력값입니다.")
        private LocalDateTime dueDate;

        public Create(String title, String content, LocalDateTime dueDate) {
            this.title = title;
            this.content = content;
            this.dueDate = dueDate;
        }

        public Todo toEntity(Member member) {
            return Todo.create(title, content, dueDate, member);
        }
    }

    @Getter
    public static class Update {
        @NotBlank(message = "제목은 필수 입력값입니다.")
        private String title;

        private final String content;

        @NotNull(message = "마감일시는 필수 입력값입니다.")
        private LocalDateTime dueDate;

        public Update(String title, String content, LocalDateTime dueDate) {
            this.title = title;
            this.content = content;
            this.dueDate = dueDate;
        }
    }

    @Getter
    public static class UpdateStatus {
        @NotNull(message = "완료 상태는 필수 입력값입니다.")
        private Boolean completed;

        public UpdateStatus(Boolean completed) {
            this.completed = completed;
        }
    }

    @Getter
    public static class ConvertRoutine {
        @NotNull(message = "루틴 ID는 필수 입력값입니다.")
        private Long routineId;

        @NotNull(message = "대상 날짜는 필수 입력값입니다.")
        private java.time.LocalDate targetDate;

        public ConvertRoutine(Long routineId, java.time.LocalDate targetDate) {
            this.routineId = routineId;
            this.targetDate = targetDate;
        }
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/todo/TodoRequest.kt
package com.memory.dto.todo

import com.memory.domain.member.Member
import com.memory.domain.todo.Todo
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDate
import java.time.LocalDateTime

class TodoRequest {

    data class Create(
        @field:NotBlank(message = "제목은 필수 입력값입니다.")
        val title: String,

        val content: String?,

        @field:NotNull(message = "마감일시는 필수 입력값입니다.")
        val dueDate: LocalDateTime
    ) {
        fun toEntity(member: Member): Todo =
            Todo.create(title, content, dueDate, member)
    }

    data class Update(
        @field:NotBlank(message = "제목은 필수 입력값입니다.")
        val title: String,

        // 자바에서 final String content → 동일하게 불변, 필요 시 null 허용
        val content: String?,

        @field:NotNull(message = "마감일시는 필수 입력값입니다.")
        val dueDate: LocalDateTime
    )

    data class UpdateStatus(
        @field:NotNull(message = "완료 상태는 필수 입력값입니다.")
        val completed: Boolean
    )

    data class ConvertRoutine(
        @field:NotNull(message = "루틴 ID는 필수 입력값입니다.")
        val routineId: Long,

        @field:NotNull(message = "대상 날짜는 필수 입력값입니다.")
        val targetDate: LocalDate
    )
}
